src/toil/batchSystems/slurm.py

```diff
@@ -344,7 +407,7 @@ def _getJobDetailsFromScontrol(self, job_id_list: list) -> dict:
                     if job_id not in job_id_list:
                         logger.log(TRACE, "%s job %d is not in the list", args[0], job_id)
                         break
-                if job_id not in job_id_list:
+                if job_id is None or job_id not in job_id_list:
```

Comment:
It's technically possible for job_id to be None if record is empty. I think this is unlikely if not impossible, but I'm not too sure what the expected behavior should be. Currently, I'm assuming it should skip the job_id.